backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/75-de.metas.fresh.kpi/5642331_sys_gh13107_add_Order_KPIs_to_dashboard.sql

```diff
@@ -0,0 +1,283 @@
+INSERT INTO WEBUI_DashboardItem (AD_Client_ID,AD_Org_ID,Created,CreatedBy,IsActive,Name,SeqNo,Updated,UpdatedBy,WEBUI_Dashboard_ID,WEBUI_DashboardItem_ID,WEBUI_DashboardWidgetType,WEBUI_KPI_ID) VALUES (1000000,1000000,TO_TIMESTAMP('2022-04-11 10:50:38','YYYY-MM-DD HH24:MI:SS'),100,'Y','Nettosumme der Kundenaufträge für diese Woche.',50,TO_TIMESTAMP('2022-04-11 10:50:38','YYYY-MM-DD HH24:MI:SS'),100,540000,540035,'T',540037)
+;
+
+-- 2022-04-11T08:50:38.394Z
+INSERT INTO WEBUI_DashboardItem_Trl (AD_Language,WEBUI_DashboardItem_ID, Name, IsTranslated,AD_Client_ID,AD_Org_ID,Created,Createdby,Updated,UpdatedBy,IsActive) SELECT l.AD_Language, t.WEBUI_DashboardItem_ID, t.Name, 'N',t.AD_Client_ID,t.AD_Org_ID,t.Created,t.Createdby,t.Updated,t.UpdatedBy,'Y' FROM AD_Language l, WEBUI_DashboardItem t WHERE l.IsActive='Y'AND (l.IsSystemLanguage='Y') AND t.WEBUI_DashboardItem_ID=540035 AND NOT EXISTS (SELECT 1 FROM WEBUI_DashboardItem_Trl tt WHERE tt.AD_Language=l.AD_Language AND tt.WEBUI_DashboardItem_ID=t.WEBUI_DashboardItem_ID)
+;
+
+-- 2022-04-14T06:48:42.485Z
+UPDATE WEBUI_KPI SET IsActive='N', SQL_WhereClause=null, SQL_From='de_metas_fresh_kpi.KPI_Order_Total_Amounts_Between_Dates(true, (NOW() - INTERVAL ''1 days'')::date, NOW()::Date, @AD_Client_ID/-1@, @AD_Org_ID/0@)',Updated=TO_TIMESTAMP('2022-04-14 08:48:42','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE WEBUI_KPI_ID=540036
+;
+
+-- 2022-04-14T06:48:56.948Z
+UPDATE WEBUI_KPI_Field SET SQL_Select='TotalOrderAmt',Updated=TO_TIMESTAMP('2022-04-14 08:48:56','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE WEBUI_KPI_Field_ID=540057
+;
+
+-- 2022-04-14T06:48:56.948Z
+UPDATE WEBUI_KPI_Field SET SQL_Select='IsoCode',Updated=TO_TIMESTAMP('2022-04-14 08:48:56','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE WEBUI_KPI_Field_ID=540058
+;
+
+-- 2022-04-14T06:49:11.457Z
+UPDATE WEBUI_KPI SET IsActive='N', SQL_WhereClause=null, SQL_From='de_metas_fresh_kpi.KPI_Order_Total_Amounts_Between_Dates(true, (NOW() - (DATE_PART(''dow'', NOW()) - 1) * ''1 day''::interval)::date, NOW()::Date, @AD_Client_ID/-1@, @AD_Org_ID/0@)',Updated=TO_TIMESTAMP('2022-04-14 08:49:11','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE WEBUI_KPI_ID=540037
+;
+
+-- 2022-04-14T06:49:21.143Z
+UPDATE WEBUI_KPI_Field SET SQL_Select='TotalOrderAmt',Updated=TO_TIMESTAMP('2022-04-14 08:49:21','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE WEBUI_KPI_Field_ID=540059
+;
+
+-- 2022-04-14T06:49:21.143Z
+UPDATE WEBUI_KPI_Field SET SQL_Select='IsoCode',Updated=TO_TIMESTAMP('2022-04-14 08:49:21','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE WEBUI_KPI_Field_ID=540060
+;
+
+-- 2022-04-14T06:49:33.860Z
+UPDATE WEBUI_KPI SET IsActive='N', SQL_WhereClause=null, SQL_From='de_metas_fresh_kpi.KPI_Order_Total_Amounts_Between_Dates(false, (NOW() - (DATE_PART(''dow'', NOW()) - 1) * ''1 day''::interval)::date, NOW()::Date, @AD_Client_ID/-1@, @AD_Org_ID/0@)',Updated=TO_TIMESTAMP('2022-04-14 08:49:33','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE WEBUI_KPI_ID=540038
```

Comment:
i'm trying to figure out how the "revenue for this week" works, but i can't. is this happening here?
 pls document it. feel free to add the SQL somewhere, or maybe even better create a dedicated little "KPI" function to do the job - maybe in a new schema called de_metas_kpi?

Reply:
we already have de_metas_fresh_kpi so added it there. Created `KPI_Order_Total_Amounts_For_Current_Week`